### PR TITLE
Feature: Add NetworkManager resource support

### LINKED ISSUE #7703: Feature - Add basic support for Network Manager. Global and Core Networks
Would be great to mock network manager resources and basic api endpoints.
https://docs.aws.amazon.com/networkmanager/latest/APIReference/Welcome.html

For example:

- CreateGlobalNetwork
- CreateCoreNetwork
- Delete*
- GetCoreNetwork
- DescribeGlobalNetworks
- TagResource
- UntagResource

### PR DESCRIPTION
**Pull request by zkarpinski**
Resolves #7703 

Adds support for NetworkManager with the following endpoints

- CreateGlobalNetwork
- DescribeGlobalNetworks
- CreateCoreNetwork
- DeleteCoreNetwork
- ListCoreNetworks
- GetCoreNetwork
- TagResource
- UntagResource

### CODE REVIEW

**Comment by zkarpinski** [comment]
@bblommers Done. Let me know if there are any other changes needed. First contribution to moto

**Comment by zkarpinski** [comment]
@bblommers Merged the latest master and fixed some issues however I'm not sure what is causing the 404 server error for Tag & Untag. My local tests work as expected.

*First contribution to moto

**Review by bblommers — APPROVED** [review]
This looks great @zkarpinski - thank you so much for adding this!

The failing tests is probably because the URL's need some manual mangling - I'll see if I can fix that now.

File: moto/networkmanager/urls.py [hunk 1]
```diff
@@ -14,4 +14,5 @@
     "{0}/global-networks/(?P<networkid>[^/.]+)$": NetworkManagerResponse.dispatch,
     "{0}/tags$": NetworkManagerResponse.dispatch,
     "{0}/tags/(?P<resourcearn>[^/.]+)$": NetworkManagerResponse.dispatch,
+    "{0}/tags/(?P<arn_prefix>[^/]+)/(?P<resource_id>[^/]+)$": NetworkManagerResponse.dispatch,
```
Comment (zkarpinski) [inline moto/networkmanager/urls.py:17]: I see, the slash in the ARN was messing with the URL. Overlooked this details.
Interesting the non-servermode works Thanks!

**Pull request merged**